### ISSUE TITLE
Separating and expanding out section on Software Architecture

### ISSUE THREAD

**Issue by OscarSeip** [carpentries-incubator/python-intermediate-development-earth-sciences#43]
Suggestion to create a separate episode for "Software Architecture" in "Introduction to Our Software Project". It appears as an important section but different from the other content. If separated, this would allow adding more context to "Separations of Concern" which seems an important point that could be introduced with more context. 

Adding the page link here for reference: https://carpentries-incubator.github.io/python-intermediate-development-earth-sciences/11-software-project/index.html

**Comment by douglowe** [comment]
I agree with the idea behind the suggestion - but we are limited by time for teaching that section, and it's possible that some that some of the extra detail would be covered later in the week anyway.

We also want to stick reasonably close to the content of the original course. @anenadic - we perhaps should consider this suggestion for the original course instead, and if changes are made to that then carry these across to our course material here?
